overlord/hookstate/ctlcmd/services_test.go

```diff
@@ -481,10 +481,11 @@ func (s *servicectlSuite) TestQueuedCommands(c *C) {
 
 	for _, ts := range tts {
 		tsTasks := ts.Tasks()
-		// assumes configure task is last
-		task := tsTasks[len(tsTasks)-1]
+
+		// default-configure hook runs before start-snap-services which runs before the configure hook
+		task := tsTasks[len(tsTasks)-4]
 		c.Assert(task.Kind(), Equals, "run-hook")
-		setup := &hookstate.HookSetup{Snap: "test-snap", Revision: snap.R(1), Hook: "configure"}
+		setup := &hookstate.HookSetup{Snap: "test-snap", Revision: snap.R(1), Hook: "default-configure"}
```

Comment:
it's strange to change preexisting tests instead of adding new ones, you can also write a shared helper that takes the hook and does the shared part between the two kind of tests

Reply:
Thanks for pointing this out, I refactored the test into two test with a shared helper

Reply:
thanks for the refactor